src/lattice_reaction_system_diffusion.jl

```diff
@@ -0,0 +1,501 @@
+### Diffusion Reaction Structure. ###
+
+# Implements the diffusionparameter metadata field.
+struct DiffusionParameter end
+Symbolics.option_to_metadata_type(::Val{:diffusionparameter}) = DiffusionParameter
+
+isdiffusionparameter(x::Num, args...) = isdiffusionparameter(Symbolics.unwrap(x), args...)
+function isdiffusionparameter(x, default = false)
+    p = Symbolics.getparent(x, nothing)
+    p === nothing || (x = p)
+    Symbolics.getmetadata(x, DiffusionParameter, default)
+end
+
+# Abstract spatial reaction structures.
+abstract type AbstractSpatialReaction end
+
+# A diffusion reaction. These are simple to hanlde, and should cover most types of spatial reactions.
+# Currently only permit constant rates.
+struct DiffusionReaction <: AbstractSpatialReaction
+    """The rate function (excluding mass action terms). Currently only constants supported"""
+    rate::Num
+    """The species that is subject to difusion."""
+    species::Num
+    """A symbol representation of the species that is subject to difusion."""
+    species_sym::Symbol    # Required for identification in certain cases.
+
+    # Creates a diffusion reaction.
+    function DiffusionReaction(rate::Num, species::Num)
+        new(rate, species, ModelingToolkit.getname(species))
+    end
+    function DiffusionReaction(rate::Number, species::Num)
+        new(Num(rate), species, ModelingToolkit.getname(species))
+    end
+    function DiffusionReaction(rate::Symbol, species::Num)
+        new(Symbolics.variable(rate), species, ModelingToolkit.getname(species))
+    end
+    function DiffusionReaction(rate::Num, species::Symbol)
+        new(rate, Symbolics.variable(species), species)
+    end
+    function DiffusionReaction(rate::Number, species::Symbol)
+        new(Num(rate), Symbolics.variable(species), species)
+    end
+    function DiffusionReaction(rate::Symbol, species::Symbol)
+        new(Symbolics.variable(rate), Symbolics.variable(species), species)
+    end
+end
+# Creates a vector of DiffusionReactions.
+function diffusion_reactions(diffusion_reactions)
+    [DiffusionReaction(dr[1], dr[2]) for dr in diffusion_reactions]
+end
+# Gets the parameters in a diffusion reaction.
+ModelingToolkit.parameters(dr::DiffusionReaction) = Symbolics.get_variables(dr.rate)
+
+### Lattice Reaction Network Structure ###
+# Desribes a spatial reaction network over a graph.
+struct LatticeReactionSystem # <: MT.AbstractTimeDependentSystem # Adding this part messes up show, disabling me from creating LRSs
```

Comment:
I think the `LatticeReactionSystem` deserves its own file as well.

Your `src/spatial` folder might look like
- `lattice_eaction_system.jl`
- `ode_problem.jl`
- `jump_problem.jl`
- `utils.jl`
- `diffusion_reaction.jl`

The benefits of keeping these in one folder but separate files is easier review process in the future and easier code navigation (tree-like structure is faster to navigate than linear structure), just to name a few benefits. In tests, you can do the same thing and put spatial tests in the `tests/spatial` folder.

Reply:
I have now separated main folder and tests into separate files.